GVFS/GVFS/CommandLine/HealthVerb.cs

```diff
@@ -0,0 +1,246 @@
+            GVFSEnlistmentHealthCalculator.GVFSEnlistmentHealthData enlistmentHealthData = enlistmentHealthCalculator.CalculateStatistics(this.Directory, this.DirectoryDisplayCount);
+
+            this.PrintOutput(enlistment, enlistmentHealthData);
+        }
+
+        private void PrintOutput(
+            GVFSEnlistment enlistment,
+            GVFSEnlistmentHealthCalculator.GVFSEnlistmentHealthData enlistmentHealthData)
+        {
+            string trackedFilesCountFormatted = enlistmentHealthData.GitTrackedItemsCount.ToString("N0");
+            string placeholderCountFormatted = enlistmentHealthData.PlaceholderCount.ToString("N0");
+            string modifiedPathsCountFormatted = enlistmentHealthData.ModifiedPathsCount.ToString("N0");
+
+            // Calculate spacing for the numbers of total files
+            int longest = Math.Max(trackedFilesCountFormatted.Length, placeholderCountFormatted.Length);
+            longest = Math.Max(longest, modifiedPathsCountFormatted.Length);
+
+            // Sort the dictionary to find the most hydrated directories by percentage
+            List<ValueTuple<string, decimal>> topLevelDirectoriesByHydration = enlistmentHealthData.DirectoryHydrationLevels;
+
+            this.Output.WriteLine("\nRepository health");
+            this.Output.WriteLine("Total files in HEAD commit:           " + trackedFilesCountFormatted.PadLeft(longest) + " | 100%");
+            this.Output.WriteLine("Files managed by VFS for Git (fast):  " + placeholderCountFormatted.PadLeft(longest) + " | " + this.FormatPercent(enlistmentHealthData.PlaceholderPercentage));
+            this.Output.WriteLine("Files managed by git (slow):          " + modifiedPathsCountFormatted.PadLeft(longest) + " | " + this.FormatPercent(enlistmentHealthData.ModifiedPathsPercentage));
+
+            this.Output.WriteLine("\nTotal hydration percentage:           " + this.FormatPercent(enlistmentHealthData.PlaceholderPercentage + enlistmentHealthData.ModifiedPathsPercentage).PadLeft(longest + 7));
+
+            this.Output.WriteLine("\nMost hydrated top level directories:");
+
+            int maxDirectoryNameLength = 0;
+            foreach ((string, decimal) pair in topLevelDirectoriesByHydration)
+            {
+                maxDirectoryNameLength = Math.Max(maxDirectoryNameLength, pair.Item1.Length);
+            }
+
+            foreach ((string, decimal) pair in topLevelDirectoriesByHydration.Take(this.DirectoryDisplayCount))
+            {
+                string dir = pair.Item1.PadRight(maxDirectoryNameLength);
+                string percent = this.FormatPercent(pair.Item2);
+                this.Output.WriteLine(" " + percent + " | " + dir);
+            }
+
+            bool healthyRepo = (enlistmentHealthData.PlaceholderPercentage + enlistmentHealthData.ModifiedPathsPercentage) < MaximumHealthyHydration;
+
+            this.Output.WriteLine("\nRepository status: " + (healthyRepo ? "Healthy" : "Unhealthy"));

```

Comment:
There will be cases where we expect directories to be fully hydrated, but are not "unhealthy". I know this is a bit of a proxy / aspirational for now, but should we tweak the status names so people don't get confused if they see "unhealthy" for a directory that is OK to be highly hydrated? Maybe "OK" and "Highly Hydrated"?

Reply:
Sure that seems reasonable. I think one of the next goals I'll work towards is translating the statistics into a more meaningful and readable output. But for now "Highly Hydrated" seems less panic-inducing 😆